server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -1290,36 +1281,92 @@ private void ensureMutable() {
      * The iterator then resumes on the first node by returning the second shard and continues until all shards from
```

Comment:
Lets update the comment to reflect the new changes

Reply:
Update, thanks!